File: apps/Mobile-App/split_the_bill/lib/utils/format.dart
import 'package:intl/intl.dart';
import '../models/user.dart';

final _cadFormat = NumberFormat.currency(locale: 'en_CA', symbol: r'$');

String formatCad(int cents) {
  return _cadFormat.format(cents / 100);
}

String displayName(User user) {
  return user.name ?? user.email;
}

String formatDateUtc(String isoDate) {
  final date = DateTime.parse(isoDate);
  return DateFormat.yMMMd().format(DateTime.utc(date.year, date.month, date.day));
}

String formatRelativeTime(String isoDate) {
  final date = DateTime.parse(isoDate).toLocal();
  final diff = DateTime.now().difference(date);
  if (diff.inMinutes < 1) return 'Just now';
  if (diff.inHours < 1) return '${diff.inMinutes}m ago';
  if (diff.inDays < 1) return '${diff.inHours}h ago';
  if (diff.inDays < 7) return '${diff.inDays}d ago';
  return DateFormat.yMMMd().format(date);
}

File: apps/web/src/utils/format.ts
export function formatCad(cents: number): string {
  return new Intl.NumberFormat('en-CA', {
    style: 'currency',
    currency: 'CAD',
  }).format(cents / 100)
}

export function displayName(user: { name: string | null; email: string } | null | undefined): string {
  if (!user) {
    return 'Unknown user'
  }

  return user.name ?? user.email
}
